src/models/GoogleDriveStorage.ts

```diff
@@ -55,12 +54,30 @@ export class GoogleDriveStorage implements StorageStrategy {
 		}
 
 		console.log('File uploaded:', result.id); // Logging the file ID
-		return result.id; // Return the file ID
+		return result;
 	}
 
 	// TODO implemenmty read and delete methods
 	async retrieve(id: string): Promise<any> {
-		throw new Error('Method not implemented.');
+		try {
+			// get the file body
+			const response = await fetch(`https://www.googleapis.com/drive/v3/files/${id}?alt=media`, {
+				method: 'GET',
+				headers: new Headers({
+					Authorization: `Bearer ${this.accessToken}`,
+				}),
+			});
+
+			const result = await response.json();
+			if (!response.ok) {
+				throw new Error(result.error.message);
+			}
+
+			console.log('File retrieved:', result); // Logging the file ID
+			return result; // Return the file ID
+		} catch (error) {
+			console.error(error);
```

Comment:
will the front end get the error to show something to user?  this console log will go into node logs, we need also to let user know somehow that we could not retrieve?  or front end will get error?

Reply:
i guess front end will get nothing, null id, and can decide what to do?  but maybe we should return some error code actually?  is it forbidden, then 401, what are the likely errors we could have?  at least we should give some clue to the front end why no file, so they know what could have happened.  was it 404, not found?

Reply:
yeah i will try to find out with Amr the best way to handle errors